attest/tpm.go

```diff
@@ -354,8 +348,12 @@ func (t *TPM) NewAK(opts *AKConfig) (*AK, error) {
 	return t.tpm.newAK(opts)
 }
 
-// NewKey creates an application key certified by the attestation key.
+// NewKey creates an application key certified by the attestation key. If opts is nil
+// then DefaultConfig is used.
 func (t *TPM) NewKey(ak *AK, opts *KeyConfig) (*Key, error) {
+	if opts == nil {
```

Comment:
This breaks the following code?

```
t.NewKey(ak, &attest.KeyConfig{})
```

Probably do something like

```
var defaultConfig = &KeyConfig{
  // ...
}

// ...
if opts == nil {
    opts = &KeyConfig{}
}
if opts.Algorithm == "" && opts.Size == 0 {
    opts = defaultConfig
}
```

Reply:
Done.